impress-org/givewp-next-gen
# Feature: Form template registration

## Description



We recently [introduced](https://github.com/impress-org/givewp-next-gen/pull/18) the initial template api as it relates to customizing the front-end.  This PR accompanies that by adding a template registration system so new Templates can get registered and pull in their front-end assets. 

One of the goals for Templates, is keeping it as simple as possible.  If someone wants to create a template that just pulls in some CSS we want them to get up and running easily.  Here is a birds eye view of a template simplicity using the (_work-in-progress_) classic template as an example:

```
class ClassicFormTemplate extends FormTemplate {

    public static function id(): string
    {
        return 'classic';
    }

    public static function name(): string
    {
        return __('Classic Template', 'give');
    }

    public function css(): string
    {
        return GIVE_NEXT_GEN_URL . 'src/NextGen/DonationForm/FormTemplates/ClassicFormTemplate/template.css';
    }

    public function js(): string
    {
        return GIVE_NEXT_GEN_URL . 'src/NextGen/DonationForm/FormTemplates/ClassicFormTemplate/template.js';
    }
}

add_action('givewp_register_form_template', function (FormTemplateRegistrar $formTemplateRegistrar) {
    $formTemplateRegistrar->registerTemplate(ClassicFormTemplate::class);
});
```

## Affects



The new "classic form" was created as the new default for next gen forms.  Viewing the form should reflect some basic CSS to show the template is active.  Note - this PR is not intended to have a complete version of the _Classic Template_ - just using it for a tangible example until we're ready to build it out.

Also added phpunit here out of necessity.

## Visuals



<img width="641" alt="Screen Shot 2022-07-13 at 5 37 43 PM" src="https://user-images.githubusercontent.com/10138447/178841155-da6b0ca1-c627-4326-ae25-eeb8db423b5e.png">


## Testing Instructions



View a NextGen form and see some basic css applied.  This is the start of what will soon be the _Classic Template_.

## Pre-review Checklist
